### ISSUE TITLE
Missing option in Intermediate Throw Events

### ISSUE THREAD

**Issue by caleeli** [ProcessMaker/modeler#971]
# Missing option in Intermediate Throw Events
## An Intermediate Throw Event could throw a Message Flow to an Intermediate Catch Event or to a Start Event. This option is missing in the Intermediate Throw Event toolbar.

![image](https://user-images.githubusercontent.com/8028650/71039815-729d1a00-20fb-11ea-90a8-841ede315c37.png)

## Reproduction steps

1. Add two pools to the diagram
2. Add an Intermediate Throw Event (A) to the first pool
3. Add an Intermediate Catch Event (B) to the second pool
4. (Missing option) Try to draw a **Message Flow** from the A to B

## Expected behaviour
There should be an option in the Intermediate Throw Event toolbar to create a message flow to an Intermediate Catch Event, a Boundary Catch Event or a Start Event from a different pool.

![image](https://user-images.githubusercontent.com/8028650/71040376-daa03000-20fc-11ea-810f-8b5cd9be2073.png)

## Actual behaviour
Missing _Message Flow_ option in the Intermediate Throw Event toolbar

## Software used
Google Chrome

## Other information
Tested using ProcessMaker develop branch

**Comment by JCook21** [comment]
@caleeli thanks for reporting this. @velkymx Some more important pieces of information/clarification:

- It seems that an intermediate message throw event can only use a message flow if the event associated with the flow is in another pool. This is the way that Camunda behaves.
- Should we allow the message flow to be associated with any elements that can accept a message flow or only with intermediate message catch events?
- How should this work with the inspector config where the user declares the name of the message that's thrown? What if the user tries to draw a message flow to a catch event that has already been configured to listen to another message?

**Comment by JCook21** [comment]
I discussed these questions with @velkymx. The decision was to allow a message flow from an intermediate message throw event to connect with any kind of message catch event (including a message start event) as long as it is in another pool or lane. We decided to disregard the configuration for the message names for now. This does mean that an association could be created between a throw and catch event that are not actually connected.
